nexus/examples/config.toml

```diff
@@ -51,8 +51,9 @@ request_body_max_bytes = 1048576
 bind_address = "[::1]:12221"
 request_body_max_bytes = 1048576
 
-[deployment.subnet]
-net = "fd00:1122:3344:0100::/56"
+[deployment.internal_dns]
+type = "from_subnet"
+subnet.net = "fd00:1122:3344:0100::/56"
```

Comment:
I'll update the documentation - I believe if the how-to-run-simulated instructions are followed, and internal DNS server is already being spun up, and this value is overwritten: 65e047323ce4388e30ee87e9972ae6fa86acdc32

Reply:
I think you have updated the "quick start" pieces in that commit (thanks!) but if somebody follows the "Running the pieces by hand" section, won't they be missing the internal DNS server?

Reply:
Starting the simulated sled agent still starts a crucible pantry and an internal DNS server, as it did before this PR. It certainly isn't getting *populated* with all the records we want, but I think that's the same as before this PR -- it only adds a record for the Crucible Pantry.

I think it would certainly be an improvement to start the Internal DNS server **separately** from the simulated sled agent, and have more of the configs point to it (rather than using hard-coded addresses, as they currently do!), but I don't think that would be a small change.

(Regardless, thanks for pointing this out - it prompted me to run through all the "simulated sled agent" manually, where I patched an unrelated bug.)

Reply:
But this configuration file is also used directly in the how-to-run-simulated instructions.  It still points Nexus at a specific IP address which is not right for the user's configuration.  Right?  It seems like there's a missing step in these instructions which is to customize this file somehow?  (and the next question is whether there's any way we could avoid that -- e.g., maybe running the simulated sled agent by hand should use a fixed IP and port for the DNS server so that we can put that directly in this file?  That's what we do for the other components there.)

Reply:
I tried to tackle this in https://github.com/oxidecomputer/omicron/pull/3364/commits/e745c76b7387bd904fd2788ca3ec8060150f9fb7 - that lets callers customize the internal DNS, DNS (not http) address which can then be propagated to Nexus.

This means that the value in the Nexus config file will be respected, as long as the same address is passed to the simulated sled agent to start a DNS server listening on it too.